lib/graphql_schema.rb

```diff
@@ -24,6 +24,10 @@ def types
     @types ||= @hash.fetch('types').map{ |type_hash| TypeDefinition.new(type_hash) }.sort_by(&:name)
   end
 
+  def type_from_name(type_name)
+    @types.find { |type| type.name == type_name }
```

Comment:
Ok cool I think we're on the same page. When I read "build and memoize a hash mapping name to type instead" I imagined it being like the code example you posted. HAH (feels like this conversation just went in one big circle)